test/index.js

```diff
@@ -158,12 +224,18 @@ describe('csp middleware', function () {
       })
   })
 
-  it('throws an error reportOnly is true and there is no report-uri', function () {
+  it('throws an error when reportOnly is true and there is no report-uri', function () {
     assert.throws(function () {
       csp({ reportOnly: true })
     }, Error)
   })
 
+  it('does not throw an error when reportOnly is a function and there is no report-uri', function () {
```

Comment:
I'm imagining a scenario in which my CSP is stable, so I decide to remove the `report-uri` in order to reduce the unnecessary load on my reporting endpoint.  I don't think this should require that I must also remove my dynamic kill switch.

I actually have an even stronger opinion here. I believe there is value in allowing a report-only policy with no `report-uri`.  For example, imagine I don't yet have a reporting endpoint set up, but I want to get started on CSP as soon as possible.  I put a report-only policy on the application, and ask the web developers to report any CSP errors they notice in the console during development.  I can't do this with helmet today.  Having said that, I understand that this is somewhat of a larger change that you may not be interested in.

If it were up to me (which it most certainly is not), I would not throw an error in any of these cases, but instead simply log a warning to the console.